src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import classNames from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../../utils/searchHelper';
+import './Pagination.scss';
+
+type Props = {
+  pages: number[];
+};
+
+export const Pagination: React.FC<Props> = ({ pages }) => {
+  const [seachParams, setSearchParams] = useSearchParams();
+  const page = seachParams.get('page' || '');
+
+  const prevPage = () => {
+    return setSearchParams(getSearchWith(seachParams, {
+      page: page && String(+page - 1),
+    }));
+  };
+
+  const nextPage = () => {
+    return setSearchParams(getSearchWith(seachParams, {
+      page: !page ? '2' : String(+page + 1),
+    }));
+  };
+
+  return (
+    <div data-cy="pagination" className="pagination">
+      <button
+        className={classNames(
+          'pagination__button', {
+            'pagination__button--disabled': !page || (page && page === '1'),
+          },
+        )}
+        type="button"
+        onClick={prevPage}
+        disabled={!page || page === '1'}
+        data-cy="paginationLeft"
+      >
+        {page && page !== '1'
+          ? (<img src="./img/arrowLeft.svg" alt="prevPage" />)
+          : (
+            <img
+              src="./img/arrowLeftDisabled.svg"
+              alt="prevPageDisabled"
+            />
+          )}
+      </button>
+
+      <div className="pagination__pages">
+        {pages.length < 5
+          ? (
+            <>
+              <button
+                type="button"
+                className={classNames(
+                  'pagination__page', {
+                    'pagination__page--isActive': !page || page === '1',
+                  },
+                )}
+                onClick={() => {
+                  setSearchParams(getSearchWith(seachParams, {
+                    page: '1',
+                  }));
+                }}
+              >
+                1
+              </button>
+              {
+                (pages.slice(1)).map(n => (
+                  <button
+                    type="button"
+                    className={classNames(
+                      'pagination__page', {
+                        'pagination__page--isActive': page && +page === n,
+                      },
+                    )}
+                    key={n}
+                    onClick={() => {
+                      setSearchParams(getSearchWith(seachParams, {
```

Comment:
I think, `seachParams` is a typo, but It's absolutely understandable that you ment `searchParams` here :)